### ISSUE TITLE
Stabekk, Høvik: OsmAnd prefers trunk_link over trunk

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#6037]
OsmAnd prefers to route along trunk_link instead of taking the trunk road.

Route is from [59.9050827,10.6080885](https://www.openstreetmap.org/#map=17/59.90508/10.60809) to 59.9012711,10.5882025

![photo_2018-09-13_16-53-23](https://user-images.githubusercontent.com/1040784/45496370-92fe5c00-b775-11e8-9f47-88d3f2cffc12.jpg)

[route.gpx.txt](https://github.com/osmandapp/Osmand/files/2379834/route.gpx.txt)

Doesn't happen with GraphHopper or OSRM: https://www.openstreetmap.org/directions?engine=graphhopper_car&route=59.9051%2C10.6083%3B59.9017%2C10.5901#map=16/59.9034/10.5992

OsmAnd version 3.2.0 with up to date Norway map. Happened also with previous OsmAnd version 3.1.6 and with previous (= August) Norway map version.

Setting a via point at the trunk road works, so routing along the trunk road is definitely possible.

**Comment by naoliv** [comment]
Probably because the `trunk_link` ways are missing a `maxspeed`
Maybe osmand should user some lower default values for `*_link` ways which don't have a `maxspeed`.

**Comment by vshcherb** [comment]
I don't think it is possible to introduce this parameter in a formal way in A* algorithm, though I don't see how can tweak parameters to let it happen with missing maxspeed when maxspeed is present on adjacent roads.

**Comment by vshcherb** [comment]
Moved to https://github.com/osmandapp/Osmand/issues/9436

**Comment by vshcherb** [comment]
Same cause of the issue  in Norway, so they can be united.
The issue is pretty simply that trunk_link doesn't have any speed limit and trunk has "80kmh", by default "trunk_link/trunk" is assumed as "100kmh", so routing goes via trunk_link. Also none of the points have "give_way" so that penalty doesn't work and doesn't imply routing.

**Comment by vshcherb** [comment]
Other examples:
Heggstadmoen (63.33064, 10.35352).
Vikhammerdalen (63.42231, 10.63092).
Malvik storsenter (63.41924, 10.75631).
Hommelvik (63.41261, 10.82056).
Rotvoll (63.43062, 10.48791), something special happens here: We can see either correct or incorrect routing.
When starting from Klepp, the route is correct. But, when moving the starting point closer (e.g., 1 km away from this position) then the error occurs here too.
